### PR TITLE
Pinned Posts

### PR DESCRIPTION
**Pull request by rjungemann**

### CODE REVIEW

File: config/default.json [hunk 1]
```diff
@@ -75,5 +75,6 @@
   "chain_id": "0000000000000000000000000000000000000000000000000000000000000000",
   "address_prefix": "STM",
   "conveyor_posting_wif": false,
-  "conveyor_username": false
+  "conveyor_username": false,
+  "pinned_posts_url": "https://raw.githubusercontent.com/steemit/condenser-pinned/master/pinned-posts.json"
```
Comment (jredbeard) [inline config/default.json]: We might want this to default to false - thoughts?

File: src/server/server-html.jsx [hunk 2]
```diff
@@ -40,14 +40,6 @@ export default function ServerHTML({ body, assets, locale, title, meta }) {
                                     content={m.content}
                                 />
                             );
-                        if (m.name && m.content)
```
Comment (roadscape) [inline src/server/server-html.jsx]: Any idea what's with this? Has it been returning dupe meta's this whole time?

File: src/app/redux/FetchDataSaga.js [hunk 3]
```diff
@@ -39,7 +39,6 @@ export function* fetchState(location_change_action) {
         const username = m[1];
         yield fork(fetchFollowCount, username);
         yield fork(loadFollows, 'getFollowersAsync', username, 'blog');
-        yield fork(loadFollows, 'getFollowingAsync', username, 'blog');
```
Comment (roadscape) [inline src/app/redux/FetchDataSaga.js]: We need to keep this line

File: src/app/components/cards/PostsList.jsx [hunk 4]
```diff
@@ -147,6 +147,7 @@ class PostsList extends React.Component {
     render() {
         const {
             posts,
+            showPinned,
```
Comment (roadscape) [inline src/app/components/cards/PostsList.jsx]: `showPinned` is set but never read

File: src/app/components/cards/PostSummary.jsx [hunk 5]
```diff
@@ -383,8 +383,10 @@ export default connect(
         return {
             post,
             content,
-            pending_payout,
-            total_payout,
+            pending_payout: pending_payout
+                ? pending_payout.toString()
+                : pending_payout,
+            total_payout: total_payout ? total_payout.toString() : total_payout,
```
Comment (roadscape) [inline src/app/components/cards/PostSummary.jsx]: What is the purpose of this change? IIRC, both these props, if set, are strings already.

File: src/server/server.js [hunk 6]
```diff
@@ -279,6 +280,10 @@ if (env === 'production') {
 if (env !== 'test') {
     const appRender = require('./app_render');
     app.use(function*() {
+        // Load the pinned posts and store them on the ctx for later use. Since
+        // we're inside a generator, we can't `await` here, so we pass a promise
+        // so `src/server/app_render.jsx` can `await` on it.
+        this.pinnedPostsPromise = pinnedPosts();
```
Comment (jredbeard) [inline src/server/server.js]: nice workaround

File: config/default.json (hunk 1 above)
Comment (rjungemann) [reply config/default.json]: It'll get overridden by the env variable, but yeah, it's probably safer to not try and show pinned posts at all, if the env variable isn't set. I'll make the change.

File: config/custom-environment-variables.json [hunk 7]
```diff
@@ -70,5 +70,6 @@
   "chain_id": "SDC_CHAIN_ID",
   "address_prefix": "SDC_ADDRESS_PREFIX",
   "conveyor_posting_wif": "CONVEYOR_POSTING_WIF",
-  "conveyor_username": "CONVEYOR_USERNAME"
+  "conveyor_username": "CONVEYOR_USERNAME",
+  "pinned_posts_url": "CONVEYOR_PINNED_POSTS_URL"
```
Comment (roadscape) [inline config/custom-environment-variables.json]: Could we remove `CONVEYOR` from the env var since it doesn't seem to rely on the conveyor service at all?

Comment (jredbeard) [reply config/custom-environment-variables.json]: agree, let's just call it PINNED_POSTS_URL - sg @rjungemann ?

File: src/app/redux/FetchDataSaga.js (hunk 3 above)
Comment (rjungemann) [reply src/app/redux/FetchDataSaga.js]: Why do we need to do this twice?

File: src/server/server-html.jsx (hunk 2 above)
Comment (rjungemann) [reply src/server/server-html.jsx]: This was removed because we check it twice (why?)

File: src/app/components/cards/PostSummary.jsx (hunk 5 above)
Comment (rjungemann) [reply src/app/components/cards/PostSummary.jsx]: It depends where the post comes from. The pinned post I get from `steem-js` has these fields as numbers.

File: config/custom-environment-variables.json (hunk 7 above)
Comment (rjungemann) [reply config/custom-environment-variables.json]: Oops I changed this and I think I rolled it back. Sorry

File: src/app/redux/FetchDataSaga.js (hunk 3 above)
Comment (jredbeard) [reply src/app/redux/FetchDataSaga.js]: at a glance I thought it was the same thing also, but it is getFollowers and getFollowing

File: src/app/components/cards/PostSummary.jsx (hunk 5 above)
Comment (roadscape) [reply src/app/components/cards/PostSummary.jsx]: Gotcha

File: src/app/redux/FetchDataSaga.js (hunk 3 above)
Comment (rjungemann) [reply src/app/redux/FetchDataSaga.js]: OH MAN. Thank you. Rolling back the change

File: src/app/components/cards/PostsList.jsx (hunk 4 above)
Comment (rjungemann) [reply src/app/components/cards/PostsList.jsx]: Fixed

**Comment by roadscape** [comment]
Tested working
